example/cmesh/t8_cmesh_geometry_examples.cxx

```diff
@@ -0,0 +1,90 @@
+/*
+  This file is part of t8code.
+  t8code is a C library to manage a collection (a forest) of multiple
+  connected adaptive space-trees of general element types in parallel.
+
+  Copyright (C) 2023 the developers
+
+  t8code is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  t8code is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with t8code; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+/* Show-case several cmesh examples with curvilinear geometries. */
+
```

Comment:
Any suggestions on how to test this cmesh?

I just want to provide a tool to quickly write out theses cmeshes and associated forests. This is a service to the user who justs wants to visualize theses cmesh examples.

I personally am always looking for such things when I check out features of a software or library.

Reply:
For this particular example it is difficult to test it. But to provide a test for t8_cmesh_set_join_by_vertices on could take the cmesh, take the vertices of the cmesh, construct the connectivity using `t8_cmesh_set_join_by_vertices` and compare it to the original connectivity.

Reply:
Not sure where you are heading here. The original connectivity for the squared disk is already generated by `t8_cmesh_set_join_by_vertices`. This is why I wrote this routine. So that I don't have to break my head in figuring out the connectivity and orientation myself. Validating this again with `t8_cmesh_set_join_by_vertices` in a test doesn't make really sense, no?

Reply:
In follow-up PRs I plan to add more curved geometry examples to `example/cmesh/t8_cmesh_geometry_examples.cxx`. My idea is that a user simply runs the binary a has the files generated and can visualize them with, for example, Paraview. Nothing fancy. There is also no overhead besides this little example file ...